src/test/java/com/google/dicomwebfuse/TestUtils.java

```diff
@@ -0,0 +1,134 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.dicomwebfuse;
+
+import com.google.dicomwebfuse.dao.http.HttpClientFactory;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.http.ProtocolVersion;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.entity.BasicHttpEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.message.BasicStatusLine;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mockito;
+
+public class TestUtils {
+
+  static public HttpClientFactory prepareHttpClient(int entityCount, int maxEntities,
+      int entityLimit,
+      String entityContent, int statusCode) throws IOException {
+
+    int iterationCount = (int) Math.ceil((double) maxEntities / entityLimit);
+    List<BasicHttpEntity> basicHttpEntities = new ArrayList<>();
+    if (entityCount < entityLimit) {
+      BasicHttpEntity httpEntity = prepareHttpEntityWithContent(entityCount, entityContent);
+      basicHttpEntities.add(httpEntity);
+    } else if (entityCount == entityLimit) {
+      BasicHttpEntity httpEntity = prepareHttpEntityWithContent(entityCount, entityContent);
+      basicHttpEntities.add(httpEntity);
+      for (int i = 0; i < iterationCount; i++) {
+        BasicHttpEntity emptyHttpEntity = prepareEmptyHttpEntity();
+        basicHttpEntities.add(emptyHttpEntity);
+      }
+    } else {
+      for (int i = 0, count = entityCount; i < iterationCount + 1; i++, count -= entityLimit) {
```

Comment:
I find this control flow a bit hard to follow due to the deep nesting. Please consider extracting a utility function.

Reply:
this code from https://github.com/GoogleCloudPlatform/healthcare-api-dicom-fuse/pull/25 moved to separated file to make it reusable, do you prefer to fix it in this PR or better to create new one?

Reply:
Let's just fix here.

Reply:
done